Cargo.toml

```diff
@@ -86,6 +86,7 @@ arrow-ipc = { version = "53.1.0", default-features = false, features = [
 arrow-ord = { version = "53.1.0", default-features = false }
 arrow-schema = { version = "53.1.0", default-features = false }
 arrow-string = { version = "53.1.0", default-features = false }
+assertor = "0.0.3"
```

Comment:
This is the one i mentioned in https://github.com/apache/datafusion/issues/11270#issuecomment-2231390487 because it seems to have the best API design goals. If it stopped being maintained, it would be a loss.
I hear you on the maintainability risk, will drop.